xblock/fields.py

```diff
@@ -752,7 +766,9 @@ def __init__(self, help=None, default=UNSET, scope=Scope.content, display_name=N
                                       **kwargs)
 
     def from_json(self, value):
-        if isinstance(value, basestring):
+        if isinstance(value, six.binary_type):
+            value = value.decode('latin-1')  # guaranteed successful decode
```

Comment:
Why is `latin-1` guaranteed to decode? What if value were encoded as something else?